### PR TITLE
Readme updates.

### PR DESCRIPTION
**Pull request by lzhoucs**
**Type:** document update

**Description:**
- Added assign operations order information.
- Removed out dated `bind` information.
- Updated package name.
- Fixed a few typos.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ harnessed widget will follow the lifecycle of a widget as if it were part of a l
 instead of updates to the DOM being applied in an async fashion, the entire harness operates in a sync fashion.
 
 In order to isolate the widget, any sub widgets (`WNode`s or node generated by `w()`) within the render will be swapped out for
-special virtual DOM nodes before being sent to the virtual DOM engine for rendering.  The are custom element tag which will look
+special virtual DOM nodes before being sent to the virtual DOM engine for rendering.  They are custom element tags which will look
```
Comment (dylans) [inline README.md]: This is better grammar, though I think it highlights that They should be replaced with what is actually meant (Do you mean subwidgets or virtual DOM nodes)... any time the context is unclear when reading quickly we should be more explicit.

**Review by dylans — CHANGES_REQUESTED** [review]
one minor nit, otherwise good to go.

File: README.md (hunk 1 above)
Comment (lzhoucs) [reply README.md]: Thanks. I think "they" meant "virtual DOM nodes". I have updated it.

**Comment by lzhoucs** [comment]
Added assign order information in the readme as discussed in https://github.com/dojo/test-extras/issues/44#issuecomment-302795265

**Review by dylans — APPROVED** [review]

**Pull request merged**